### PR TITLE
prevent active-series underflow in local series limiter

### PR DESCRIPTION
**Pull request by carles-grafana**
`tempo_metrics_generator_registry_active_series` could wrap due to uint32 underflow
  when stale cleanup deletes overflow series that were never counted as active.

  - ignore `OnDelete` for overflow entity hash
  - clamp delete count to current active-series value before subtracting
  - add regression test for accepted+overflow stale-delete sequence

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by knylander-grafana** [comment]
Do we need a changelog entry for this?

File: modules/generator/localserieslimiter/local_series_limiter.go [hunk 1]
```diff
@@ -114,7 +114,19 @@ func (l *LocalSeriesLimiter) OnUpdate(uint64, uint32) {
...
-func (l *LocalSeriesLimiter) OnDelete(_ uint64, seriesCount uint32) {
+func (l *LocalSeriesLimiter) OnDelete(hash uint64, seriesCount uint32) {
+	// Overflow series are never counted in activeSeries, so deleting them must
+	// not decrement the active series counter.
+	if hash == l.overflowEntityHash {
+		return
+	}
+
+	activeSeries := l.activeSeries.Load()
+	// Clamp delete count to avoid uint32 underflow if we receive a mismatched delete.
```
Comment (oleg-kozlyuk-grafana) [inline modules/generator/localserieslimiter/local_series_limiter.go:125]: To be quite honest, I don't think this solves the issue. This basically means that a "mismatched" delete will render the active series counter completely unreliable. Is there a way to solve this by getting accurate counters of active series?

Comment (carles-grafana) [reply modules/generator/localserieslimiter/local_series_limiter.go:125]: after the overflow series fix, the count should be accurate. This is a safeguard for future bugs that could cause an underflow

Comment (carles-grafana) [reply modules/generator/localserieslimiter/local_series_limiter.go:125]: what solves the issue is the overflow check above

Comment (oleg-kozlyuk-grafana) [reply modules/generator/localserieslimiter/local_series_limiter.go:125]: If this is purely defensive, fair enough.
